### ISSUE TITLE
Build failures because many undefines because DARKMODE not defined

### ISSUE THREAD

**Issue by sharmanr** [mpogue2/SquareDesk#927]
After downgrading back to Ventura I had to re-create the github directory and re-install Qt.  When I try to build SquareDesk there are many undefines of things whose definitions are enclosed within ifdef DARKMODE.
E.g.
/Users/richard/GitHub/SquareDesk/SquareDesk-DEV/test123/playlists.cpp:1927: error: use of undeclared identifier 'updateTreeWidget'
/Users/richard/GitHub/SquareDesk/SquareDesk-DEV/test123/mainwindow.cpp:9789: error: use of undeclared identifier 'darkSelectedSongRow'; did you mean 'selectedSongRow'?
../../SquareDesk-DEV/test123/mainwindow.cpp:9789:27: error: use of undeclared identifier 'darkSelectedSongRow'; did you mean 'selectedSongRow'?

**Comment by mpogue2** [comment]
OK, let me check this out.  I probably broke it!  If you have a chance, tell me which exact commit you tried to build from!

**Comment by sharmanr** [comment]
I cd’d to  /Users/richard/Github/SquareDesk/SquareDesk-DEV/ and double clicked on SquareDesk.pro the configured the project and pressed the build button.

The output from the Compile Output window starts like this:

16:57:06: Running steps for project SquareDesk...
16:57:06: Configuration unchanged, skipping qmake step.
16:57:06: Starting: "/usr/bin/make" -j8
cd sdlib/ && ( test -e Makefile || /Users/richard/Qt/6.5.3/macos/bin/qmake -o Makefile /Users/richard/GitHub/SquareDesk/SquareDesk-DEV/sdlib/sdlib.pro -spec macx-clang CONFIG+=debug CONFIG+=qml_debug ) && /Applications/Xcode.app/Contents/Developer/usr/bin/make -f Makefile 
cd taglib/taglib/ && ( test -e Makefile || /Users/richard/Qt/6.5.3/macos/bin/qmake -o Makefile /Users/richard/GitHub/SquareDesk/SquareDesk-DEV/taglib/taglib/taglib.pro -spec macx-clang CONFIG+=debug CONFIG+=qml_debug ) && /Applications/Xcode.app/Contents/Developer/usr/bin/make -f Makefile 

Does that help?

**Comment by mpogue2** [comment]
Thanks for the report and the hint!  I know exactly what the problem is, and I'm testing a fix right now...

**Comment by sharmanr** [comment]
Thanks Mike!

**Comment by mpogue2** [comment]
@sharmanr OK, try now!  Top-of-tree should compile properly in DARKMODE and non-DARKMODE now.

Note that the Vamp executables are built separately (via the Vamp package), and the two files (qm-vamp-plugins.dylib and vamp-simple-host) are just copied into the Contents/MacOS folder in your SquareDesk.app bundle in your build directory (this is a manual step right now).  These make setting loops really fast, so you might want these.

[VampExecutables.zip](https://github.com/mpogue2/SquareDesk/files/12842089/VampExecutables.zip)

**Comment by mpogue2** [comment]
Also note: Use Qt6.4.3 for now, as referenced in #924.

**Comment by sharmanr** [comment]
Got it.  Thanks.

**Comment by sharmanr** [comment]
The Qt I  re-installed yesterday gave me 6.5.3 but it worked fine.

**Comment by mpogue2** [comment]
@sharmanr You might want to reserve judgment until you've had a chance to run with it a while. :-)  The flashing problem with 6.5.2 and 6.5.3 is intermittent.  When it happens, it's super annoying, and I can't see us releasing anything until we figure that out.  

Near as I can tell, it's associated with update()'s, because when it flashes slowly, it flashes at the same time as the second hand on the clock moves.  Other times, flashing is nearly continuous, especially after starting playback (when the VU meter timer is running).  

Please let me know how you're testing it, such that you are NOT seeing this problem, because I'm seeing it almost every time with:

- M1 MacBook Air running Ventura
- external wide-screen monitor

When it happens, reverting back to Qt6.4.3 always makes the problem go away for me.

**Comment by mpogue2** [comment]
Hmmm...the problem only seems to start when I move the SquareDesk window between the big monitor screen and the laptop screen.  If I don't move it, it seems pretty stable.  That is WEIRD.

**Comment by sharmanr** [comment]
What is the flashing problem?  I couldn't find it in open issues.
Assuming that the problem is obvious when it happens(!) I haven't hit it so far.  This is with Xcode 15 Qt 6.5.3 on M2 desktop.  I copied the .dmg file and installing it on an M1 Macbook also running Ventura all seems well but have had limited testing.    I tried to build there (which would have been Qt 6.6.0 and Xcode 14.3) but ran into a build error so will retry later.

**Comment by mpogue2** [comment]
@sharmanr Flashing problem was partially documented in #924.  Then when I moved to 6.5.3, I saw it again.  It seems to start when I move the window between my big monitor and my laptop monitor.  Once it starts, it can't be stopped.  Sometimes the flashing is slow, and when it is, it's always synchronous to when the second hand on the clock moves (therefore, perhaps an update() problem?)

If I don't move the window between monitors, it seems that with 6.5.3 the problem never occurs.  Weird, but I guess it's good enough to switch to 6.5.3 for now...

Here's a GIF here to show you:

![flashingProblem_20231010T 220057](https://github.com/mpogue2/SquareDesk/assets/6316003/2d086db3-dc59-4d56-bb08-3877ca9b01dd)
